### ISSUE TITLE
Translated Emails

### ISSUE THREAD

**Issue by lukevella** [lukevella/rallly#547]
Previously this was quite hard to do because the email templates where precompiled html files but now that we are using React as a templating engine for emails it should be possible to use `react-i18next` to translate these emails as well.

The subject lines will need a separate solution (hopefully using `i18next`) since they are not part of the templates.

<sub>[RAL-247](https://linear.app/rallly/issue/RAL-247/translated-emails)</sub>

## Upvote & Fund

- We're using [Polar.sh](https://polar.sh/lukevella) so you can upvote and help fund this issue.
- We receive the funding once the issue is completed & confirmed by you.
- Thank you in advance for helping prioritize & fund our backlog.

<a href="https://polar.sh/lukevella/rallly/issues/547">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/lukevella/rallly/issues/547/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/lukevella/rallly/issues/547/pledge.svg">
</picture>
</a>

**Comment by ltguillaume** [comment]
Any progress on this? 🙂

Since emails are an integral part of the workflow, unfortuantely there are quite a few people I cannot use this for if the emails aren't translated.

**Comment by ltguillaume** [comment]
@lukevella I think this is really essential for the people to not get confused, so if you'd like, perhaps I could create a PR for this, provided that I'm given some pointers on how this translation framework is implemented.

**Comment by lukevella** [comment]
Just want to add that for this feature to be delivered we need to be able to know the users preferred locale on the server side. One way to do this is to include the locale as a parameter for each function that could trigger an email. This can be quite cumbersome and is error-prone. Another option is to modify how user sessions work such that a locale is always set. Currently it's only set when explicitly modified by the user. I'd much prefer the latter option but it's a significant piece of work.

**Comment by lukevella** [comment]
That only works for new users and I believe is partially done. We still need a way to get this info from existing users and even guests that don't have an account can receive emails which should be in their preferred language.

**Comment by lukevella** [comment]
Looks like you updated your previous comment with more info. Thanks for that.

I'm not sure what your goal is with your comments. Would you like me to say that this is not a significant piece of work? Simply writing down ideas doesn't solve the problem.

**Comment by lukevella** [comment]
I only said that it is significant. It merits having its own issue which I intend to create in due time.

As you might have noticed, when you start pulling this thread you will find various different cases that need to be handled that are not immediately obvious.

My comment was a note to myself and to anyone that cares that there is slightly more to this task than being able to render emails in different languages.

**Comment by melegiul** [comment]
Hi. First of all thank for providing this awesome tool. I really like it. :tada: 

I know, you recently just removed the good first issue label. But do you think you would still accept PRs from the community?

**Comment by lukevella** [comment]
@melegiul, possibly yes! If you're interested, maybe reach out on Discord and we can discuss the approach before starting any work or submitting a PR.

**Comment by lukevella** [comment]
Closed by #1278. Translations can be submitted through [Crowdin](https://crowdin.com/project/rallly)

**Comment by lukevella** [comment]
Hi @ltguillaume, the translation text is correct here because `<domain />` gets replaced with a link to the domain where the email was sent from. I understand this is a bit confusing since we treat support email differently so will consider amending this at some point.
